src/commands/info.js

```diff
@@ -0,0 +1,252 @@
+import { BusyIndicator } from '../util/busyindicator.js';
+import chalk from 'chalk';
+import { detect } from '../util/detect.js';
+import wrapAnsi from 'wrap-ansi';
+import { basename } from 'node:path';
+
+const { bold, cyan, gray, magenta, red, yellow } = chalk;
+const typesList = ['all', 'os', 'nodejs', 'titanium', 'jdk', 'android', 'ios'];
+
+/**
+ * Returns the configuration for the info command.
+ * @param {Object} logger - The logger instance
+ * @param {Object} config - The CLI config object
+ * @param {CLI} cli - The CLI instance
+ * @returns {Object} Info command configuration
+ */
+export function config(_logger, _config, _cli) {
+	return {
+		title: 'Info',
+		skipBanner: true,
+		flags: {
+			json: {
+				desc: 'display info as json'
+			}
+		},
+		options: {
+			output: {
+				abbr: 'o',
+				default: 'report',
+				hidden: true,
+				values: ['report', 'json']
+			},
+			types: {
+				abbr: 't',
+				default: 'all',
+				desc: 'information types to display; you may select one or more',
+				values: typesList.filter(t => t !== 'ios' || process.platform === 'darwin')
+			}
+		}
+	};
+}
+
+/**
+ * Displays information about the current system.
+ * @param {Object} logger - The logger instance
+ * @param {Object} config - The CLI config object
+ * @param {CLI} cli - The CLI instance
+ * @param {Function} finished - Callback when the command finishes
+ */
+export async function run(logger, config, cli) {
+	let busy;
+	const isJson = cli.argv.json || cli.argv.output === 'json';
+
+	if (!isJson) {
+		logger.skipBanner(false);
+		logger.banner();
+		if (!cli.argv.quiet && !!cli.argv['progress-bars']) {
+			busy = new BusyIndicator();
+			busy.start();
+		}
+	}
+
+	// determine the types to display
+	const types = {};
+	let i = 0;
+	const specifiedTypes = (cli.argv.types || 'all').toLowerCase().split(',');
+	for (let t of specifiedTypes) {
+		t = t.trim();
+		if (typesList.indexOf(t) >= 0) {
+			types[t] = ++i;
+		}
+	}
+	if (i === 0) {
+		types.all = 1;
+	}
+
+	let data;
+	let platformInfo;
+	try {
+		({
+			data,
+			platformInfo
+		} = await detect(logger, config, cli, types));
+	} finally {
+		busy?.stop();
+	}
+
+	if (isJson) {
+		logger.log(JSON.stringify(data, null, '\t'));
+		process.exit(0);
```

Comment:
This should return rather than exit as with the info being a large string it doesn't get flushed to stdout before process.exit happens (writing to stdout is async). Running the below demonstrates this.

```js
import { spawn } from 'child_process';
let out = '';
const process = spawn('ti', ['info', '-o', 'json']);
process.stdout.on('data', (d) => out += d);
process.on('close', () => console.log(out));
```

```suggestion
		return
```

I imagine throughout the CLI probably rather than calling `process.exit` we would want to set `process.exitCode` and return?

Reply:
I thought I fixed this. Why does the unit test work? I'll fix.